core/deployment/src/test/java/io/quarkiverse/langchain4j/test/ToolProviderTest.java

```diff
@@ -0,0 +1,122 @@
+import org.jboss.shrinkwrap.api.ShrinkWrap;
+import org.jboss.shrinkwrap.api.spec.JavaArchive;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.RegisterExtension;
+
+import java.util.List;
+import java.util.function.Supplier;
+
+import static dev.langchain4j.data.message.ChatMessageType.TOOL_EXECUTION_RESULT;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+
+class ToolProviderTest {
+    @Inject
+    MyServiceWithToolProvider myServiceWithTools;
+
+    @Inject
+    MyServiceWithoutToolProvider myServiceWithoutTools;
+
+    @ApplicationScoped
+    public static class MyCustomToolProvider implements ToolProvider {
+        @Inject
+        MyServiceWithoutToolProvider myServiceWithoutTools;
+
+        @Override
+        public ToolProviderResult provideTools(ToolProviderRequest request) {
+            assertNotNull(myServiceWithoutTools);
+            
+            ToolSpecification toolSpecification = ToolSpecification.builder()
+                    .name("get_booking_details")
+                    .description("Returns booking details")
+                    .build();
+            ToolExecutor toolExecutor = (t, m) -> "0";
+            return ToolProviderResult.builder()
+                    .add(toolSpecification, toolExecutor)
+                    .build();
+        }
+    }
+
+    public static class TestAiSupplier implements Supplier<ChatLanguageModel> {
+        @Override
+        public ChatLanguageModel get() {
+            return new TestAiModel();
+        }
+    }
+
+    public static class TestAiModel implements ChatLanguageModel {
+        @Override
+        public Response<AiMessage> generate(List<ChatMessage> messages) {
+            return new Response<>(new AiMessage("42"));
+        }
+
+        @Override
+        public Response<AiMessage> generate(List<ChatMessage> messages, List<ToolSpecification> toolSpecifications) {
+            ChatMessage lastMsg = messages.get(messages.size() - 1);
+            boolean isLastMsgToolResponse = lastMsg.type().equals(TOOL_EXECUTION_RESULT);
+            if (isLastMsgToolResponse) {
+                ToolExecutionResultMessage msg = (ToolExecutionResultMessage) lastMsg;
+                return new Response<>(new AiMessage(msg.text()));
+            }
+            ToolSpecification toolSpecification = toolSpecifications.get(0);
+            ToolExecutionRequest toolExecutionRequest = ToolExecutionRequest.builder()
+                    .name(toolSpecification.name())
+                    .id(toolSpecification.name())
+                    .build();
+            TokenUsage usage = new TokenUsage(42, 42);
+            return new Response<>(AiMessage.from(toolExecutionRequest), usage, FinishReason.TOOL_EXECUTION);
+        }
+    }
+
+    @RegisterAiService(toolProvider = MyCustomToolProvider.class, chatLanguageModelSupplier = TestAiSupplier.class)
+    interface MyServiceWithToolProvider {
```

Comment:
I think the name here is a little misleading. To me, it makes to call it  `MyServiceWithCustomToolProvider`